### ISSUE TITLE
CNI integration

### ISSUE THREAD

**Issue by richardcase** [weaveworks/flintlock#14]
Integrate CNI into flintlock so that the network interfaces on the host are created/configured via CNI. We need to expose this via the API and also provide a default config.

**Comment by yitsushi** [comment]
https://github.com/kubevirt/macvtap-cni

**Comment by richardcase** [comment]
Yeah there are a few plugins. With #13 we need to look at what plugins there are for macvtap and come up with a recommendation. 

When i looked back in April there was a difference in functionality and activity in the plugins i found.

**Comment by richardcase** [comment]
This is still needed.

**Comment by richardcase** [comment]
Yes this is something that we still want / need to do.

**Comment by Callisto13** [comment]
https://youtu.be/Kvdh78TURck

**Comment by yitsushi** [comment]
Still valid.

**Comment by pablochacin** [comment]
Interested

**Comment by richardcase** [comment]
@pablochacin - we currently use netlink directly to create the interfaces (tap or macvtap).  We also support attaching to a bridge.

The Firecracker Go SDK support using CNI. However, thats good for Firecracker but won't cover Cloud Hypervisor. So we probably need to integrate with CNI directly in Flintlock.

**Comment by pablochacin** [comment]
Hi @richardcase Thanks for the update.

**Comment by pablochacin** [comment]
@richardcase I'm revisiting this issue.

 I noticed https://github.com/liquidmetal-dev/flintlock/issues/856 
 
 Does it affect the scope of this issue? My guess is that not, but wanted to check your thought about this.

**Comment by richardcase** [comment]
I don't think it affects the scope of the issue. We are going to remove macvtap from the firecracker side as we can't keep using our  own fork of Firecracker.

Let me know if i can help with anything.

**Comment by pablochacin** [comment]
@richardcase  can please assign it to me?

**Comment by richardcase** [comment]
Thanks @yitsushi , all yours @pablochacin
